src/network/netdev/batadv.c

```diff
@@ -35,6 +35,42 @@ static void batadv_init(NetDev *n) {
         b = BATMAN(n);
 }
 
+int config_parse_batadv_gateway_mode(const char *unit,
+                                                   const char *filename,
+                                                   unsigned line,
+                                                   const char *section,
+                                                   unsigned section_line,
+                                                   const char *lvalue,
+                                                   int ltype,
+                                                   const char *rvalue,
+                                                   void *data,
+                                                   void *userdata) {
+
+        Batadv *b = userdata;
+        int mode;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        mode = BATADV_GW_MODE_OFF;
+
+        if (strcmp("client", rvalue) == 0)
+                mode = BATADV_GW_MODE_CLIENT;
+
+        else if (strcmp("server", rvalue) == 0)
+                mode = BATADV_GW_MODE_SERVER;
+
+        else
+                log_syntax(unit, LOG_ERR, filename, line, -EINVAL, "Batadv gateway mode '%s' is not valid. Value"
```

Comment:
EINVAL is not necessary. LOG_ERR → LOG_WARN.
Please don't line-break the message, just include it on its own separate line.
Also, if the rvalue is unkown, no modification should be done.
```C
else {
        log_syntax(unit, LOG_WARN, filename, line, 0,
                          "Batadv gateway mode '%s' unknown. Ignoring.", rvalue);
        return 0;
}
```